autobahn/asyncio/component.py

```diff
@@ -350,18 +350,24 @@ def run(components, log_level='info'):
     #   signal.signal(signal.SIGINT, signal.SIG_DFL)
 
     @asyncio.coroutine
-    def exit():
-        return loop.stop()
-
     def nicely_exit(signal):
         log.info("Shutting down due to {signal}", signal=signal)
-        for task in asyncio.Task.all_tasks():
-            task.cancel()
-        asyncio.ensure_future(exit())
+        tasks = asyncio.Task.all_tasks()
+        tasks.remove(asyncio.Task.current_task())
+        [task.cancel() for task in tasks]
+
```

Comment:
This "bare" list-comprehension is a little weird (and I'm not sure it'll "do the right thing" in PyPy..?). I think it would be simpler/more-clear to make this a "for task in tasks: task.cancel()"